subworkflows/local/rose.nf

```diff
@@ -1,23 +1,102 @@
-include { ROSE as RUN_ROSE           } from "../../modules/local/rose"
-include { UCSC_GTFTOGENEPRED              } from "../../modules/nf-core/ucsc/gtftogenepred"
+include { GAWK as FILTER_CONVERT_GTF } from '../../modules/nf-core/gawk'
+include { GNU_SORT as SORT_BED } from '../../modules/nf-core/gnu/sort'
+include { BEDTOOLS_SLOP as CONSTRUCT_TSS } from '../../modules/nf-core/bedtools/slop'
+include { BEDTOOLS_SUBTRACT as FILTER_PREDICTIONS } from '../../modules/nf-core/bedtools/subtract'
+include { BEDTOOLS_COMPLEMENT as INVERT_TSS } from '../../modules/nf-core/bedtools/complement'
+include { BEDTOOLS_MERGE as STITCHING } from '../../modules/nf-core/bedtools/merge'
+include { BEDTOOLS_INTERSECT as TSS_OVERLAP } from '../../modules/nf-core/bedtools/intersect'
+include { GAWK as FILTER_OVERLAPS } from '../../modules/nf-core/gawk'
+include { BEDTOOLS_SUBTRACT as SUBTRACT_OVERLAPS } from '../../modules/nf-core/bedtools/subtract'
+include { BEDTOOLS_INTERSECT as UNSTITCHED_REGIONS } from '../../modules/nf-core/bedtools/intersect'
+include { GNU_SORT as CONCAT_AND_SORT } from '../../modules/nf-core/gnu/sort'
 
 workflow ROSE {
     take:
     ch_bed
     ch_gtf
+    chrom_sizes
 
     main:
 
     ch_versions = Channel.empty()
 
-    UCSC_GTFTOGENEPRED(ch_gtf)
-    RUN_ROSE(ch_bed, UCSC_GTFTOGENEPRED.out.genepred)
+    // Convert GTF to BED format and collapse regions to a single base pair at their start positions
+    FILTER_CONVERT_GTF(ch_gtf, [])
 
-    ch_versions = ch_versions.mix(RUN_ROSE.out.versions)
-    ch_versions = ch_versions.mix(UCSC_GTFTOGENEPRED.out.versions)
+    // Downstream methods require sorted inputs
+    SORT_BED(FILTER_CONVERT_GTF.out.output)
+
+    // Construct 5000 bp window around transcription start site (TSS)
+    CONSTRUCT_TSS(SORT_BED.out.sorted, chrom_sizes.map{meta, file -> file})
+
+    INVERT_TSS(CONSTRUCT_TSS.out.bed, chrom_sizes.map{meta, file -> file})
+
+    ch_enhancers = ch_bed
+        .filter{meta, bed -> meta.assay.contains("enhancers")}
+        .combine(CONSTRUCT_TSS.out.bed)
+        .map{meta1, pred, meta2, tss -> [meta1, pred, tss]}
+
+    ch_promoters = ch_bed
+        .filter{meta, bed -> meta.assay.contains("promoters")}
+        .combine(INVERT_TSS.out.bed)
+        .map{meta1, pred, meta2, non_tss -> [meta1, pred, non_tss]}
```

Comment:
This should be done using the [branch operator](https://www.nextflow.io/docs/latest/operator.html#branch)